### DISCUSSION TITLE
Multi tenancy support?

### DISCUSSION THREAD

**Discussion by RobTF** [simpleidserver/SimpleIdServer#734] [Q&A]
Hi,

I'd like to allow multiple customers to use SCIM to sync their groups/users. I think I'd need the following;

- A way to dynamically provision API keys per customer.
- A way to mark each provisioned entity as belonging to one of the customers, determined by their API key.

This is possible out of the box? If not, is there any guidance on how this could be added to the current architecture of the server? For example implementing a custom `ApiKeyProvider`?

**Comment by simpleidserver** [comment]
Hello,

As this feature has been requested multiple times in the project, and I have proposed temporary solutions to support multi-tenancy in SCIM, I create a new ticket #736 to add multi-tenant support in the SCIM server.

Kind regards,

SID

**Reply by RobTF** [reply]
Thanks - I'm working around it for now similarly to how one other user is whereby I'm adding global filters and interceptors to EF Core to add and populate the tenant column alongside a middleware which configures a tenant ID based on the URI.

**Reply by simpleidserver** [reply]
Hello,

I have made some modifications in the master branch to support multi-tenancy in the SCIM server.

The feature can be enabled by editing the appsettings.json file and setting the property IsRealmEnabled to true. By default, there are two tenants/realms configured in the applicatio


| Tenant | Owner | API key | URL |
| ------ | ----- | ------- | --- |
| idserver | IdServer | ba521b3b-02f7-4a37-b03c-58f713bf88e7 | https://localhost:5003/idserver |
| azuread | AzureAd | 1595a72a-2804-495d-8a8a-2c861e7a736a | https://localhost:5003/azuread |

You can add a new one by inserting a row in the Realm table with the following values:
* Name: name of the realm
* Owner: NameIdentifier of the user. Possible values are listed in the property ApiKeyConfigurations.ApiKeys.Owner of the appsettings.json file.

KR,

SID
